### ISSUE TITLE
Support multi-SRPM builds in rpmdeps inspection

### ISSUE THREAD

**Issue by jchecahi** [rpminspect/rpminspect#1246]
**Describe the bug**
Running `rpmdeps` inspection on modules can result in false positives regarding explicit package version requirements for shared libraries. This will happen because modules are generally a grouping of several packages, hence containing subpackages coming from different SRPMs. However, the inspection assumes that all the subpackages come from the same SRPM. Hence, it may infers that two subpackages belong to the same package when they don't.

Specific use case: llvm-toolset module. `clang-libs` subpackage from `clang` package needs `libLLVM-XX.so` (`XX` being LLVM major version). This is provided by `llvm-libs` subpackage from `llvm` package:
```
Subpackage clang-libs on aarch64 carries 'Requires: libLLVM-16.so()(64bit)' which comes
from subpackage llvm-libs but does not carry an explicit package version requirement.
Please add 'Requires: llvm-libs = %{version}-%{release}' to the spec file to avoid the
need to test interoperability between various combinations of old and new subpackages. 

Result: VERIFY
```
However rpminspect should not report this as a missing explicit requires, as it's fine to not have `clang-libs` tied to the same `%{version}-%{release}` as `llvm-libs`, because they come from different packages. Also the [packaging guidelines](https://docs.fedoraproject.org/en-US/packaging-guidelines/#_explicit_requires) advise to not add unnecessary explicit requires

**To Reproduce**
Using aarch64 only for faster testing:
```
rpminspect-redhat -v -T rpmdeps --arch aarch64 llvm-toolset-rhel8-8090020230803191529.b46abd14
```

**Expected behavior**
rpminspect should not report shared library dependencies from a subpackage as "missing explicit requires" when the shared library is provided by a subpackage coming from a different SRPM.

**Comment by dcantrell** [comment]
I am unable to reproduce this problem with the current source, so I think this has been resolved.  The rpmdeps inspection passes now for the module build in this example and I do remember working on it in the past but perhaps I forgot to close the issue.
